### PR TITLE
Fixes regarding default delivery address

### PR DESCRIPTION
**Pull request by sebaholesz**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Various things regarding default delivery address were fixed. Check commit messages for more info.
|New feature| No
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No
|Fixes issues| ...
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://sh-default-delivery-address-in-order.odin.shopsys.cloud
  - https://cz.sh-default-delivery-address-in-order.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/hooks/user/useCurrentUserContactInformation.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const mergeContactInformation = (
         const filteredProperty = filteredContactInformationFromStore[key as keyof ContactInformation];
 
         const isEmptyString = typeof filteredProperty === 'string' && filteredProperty.length === 0;
-        if (isEmptyString && key in contactInformationFromApi) {
+        if ((isEmptyString || filteredProperty === null) && key in contactInformationFromApi) {
```
Comment (tvikito) [inline project-base/storefront/hooks/user/useCurrentUserContactInformation.ts]: Můžeme odebrat check pro `null`?

```suggestion
        if ((isEmptyString || !filteredProperty) && key in contactInformationFromApi) {
```

**Review by tvikito — APPROVED** [review]

File: project-base/storefront/hooks/user/useCurrentUserContactInformation.ts (hunk 1 above)
Comment (sebaholesz) [reply project-base/storefront/hooks/user/useCurrentUserContactInformation.ts]: I don't think it makes sense. We do not want to filter out all falsy values, just empty strings and nulls.

Comment (tvikito) [reply project-base/storefront/hooks/user/useCurrentUserContactInformation.ts]: Ah, ok. It was not obvious from code here. But I've checked and I see also `undefined` as possible value, don't you want to include it as well?

Comment (sebaholesz) [reply project-base/storefront/hooks/user/useCurrentUserContactInformation.ts]: I did, thanks for the suggestion

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**